collectors/cgroups.plugin/sys_fs_cgroup.c

```diff
@@ -2462,63 +2498,150 @@ static inline void share_cgroups()
+    }
+
+    if (matches_enabled_cgroup_renames(cg->id)) {
+        debug(D_CGROUP, "cgroup '%s' (name '%s') matches 'run script to rename cgroups matching', will try to rename it", cg->id, cg->chart_title);
+        if (is_inside_k8s && k8s_is_container(cg->id)) {
+            // it may take up to a minute for the K8s API to return data for the container
+            // tested on AWS K8s cluster with 100% CPU utilization
+            cg->pending_renames = 9; // 1.5 minute
+        } else {
+            cg->pending_renames = 2;
         }
+    }
+}
 
-        if(cgroup_search_in_devices) {
-            if(find_dir_in_subdirs(cgroup_devices_base, NULL, found_subdir_in_dir) == -1) {
-                cgroup_search_in_devices = 0;
-                error("CGROUP: disabled devices statistics.");
-            }
+static inline void discovery_process_cgroup(struct cgroup *cg) {
+    if (!cg->available || cg->processed) {
+        return;
+    }
+
+    if (cg->first_time_seen) {
+        discovery_process_first_time_seen_cgroup(cg);
+        if (unlikely(cg->first_time_seen || cg->processed)) {
+            return;
         }
     }
-    else {
-        if (find_dir_in_subdirs(cgroup_unified_base, NULL, found_subdir_in_dir) == -1) {
-            cgroup_unified_exist = CONFIG_BOOLEAN_NO;
-            error("CGROUP: disabled unified cgroups statistics.");
+
+    if (cg->pending_renames) {
+        discovery_rename_cgroup(cg);
+        if (unlikely(cg->pending_renames || cg->processed)) {
+            return;
         }
     }
 
-    update_filenames();
+    cg->processed = 1;
+
+    if (is_cgroup_systemd_service(cg)) {
+        cg->enabled = 1;
+        return;
+    }
+
+    if (!(cg->enabled = matches_enabled_cgroup_names(cg->chart_title))) {
+        debug(D_CGROUP, "cgroup '%s' (name '%s') disabled by 'enable by default cgroups names matching'", cg->id, cg->chart_title);
+        return;
+    }
+
+    if (!(cg->enabled = matches_enabled_cgroup_paths(cg->id))) {
+        debug(D_CGROUP, "cgroup '%s' (name '%s') disabled by 'enable by default cgroups matching'", cg->id, cg->chart_title);
+        return;
+    }
+
+    if (cg->enabled) {
+        read_cgroup_network_interfaces(cg);
+    }
+}
+
+static inline void discovery_find_all_cgroups() {
+    debug(D_CGROUP, "searching for cgroups");
+
+    discovery_mark_all_cgroups_as_unavailable();
+
+    if (!cgroup_use_unified_cgroups) {
+        discovery_find_all_cgroups_v1();
+    } else {
+        discovery_find_all_cgroups_v2();
+    }
+
+    struct cgroup *cg;
+    for (cg = discovered_cgroup_root; cg; cg = cg->discovered_next) {
+        discovery_process_cgroup(cg);
+    }
+
+    discovery_update_filenames();
 
     uv_mutex_lock(&cgroup_root_mutex);
-    cleanup_all_cgroups();
-    copy_discovered_cgroups();
+    discovery_cleanup_all_cgroups();
+    discovery_copy_discovered_cgroups_to_reader();
     uv_mutex_unlock(&cgroup_root_mutex);
 
-    share_cgroups();
+    discovery_share_cgroups_with_ebpf();
```

Comment:
I agree that right now we are only sharing with eBPF, but the motive I did not add `ebpf` for the name, it is because I expect this can be shared with other plugins in the future.

Reply:
Cool, will rename the function when you will do it.